### PR TITLE
[SPARK-31393][SQL] Show the correct alias in schema for expression

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Some alias of expression can not display correctly in schema. This PR will fix them.
- `TimeWindow`
- `MaxBy`
- `MinBy`
- `UnaryMinus`
- `BitwiseCount`

This PR also fix a typo issue, please look at https://github.com/apache/spark/blob/b7cde42b04b21c9bfee6535199cf385855c15853/sql/core/src/test/scala/org/apache/spark/sql/ExpressionsSchemaSuite.scala#L142

Note: 

1. `MaxBy` and `MinBy` extends `MaxMinBy` and the latter add a method `funcName` not needed.  We can reuse `prettyName` to replace `funcName`.
2. Spark SQL exists some function no elegant implementation.For example: `BitwiseCount` override the sql method show below:
`override def sql: String = s"bit_count(${child.sql})"`
I don't think it's elegant enough. Because `Expression` gives the following definitions.
```
  def sql: String = {
    val childrenSQL = children.map(_.sql).mkString(", ")
    s"$prettyName($childrenSQL)"
  }
```
By this definition, `BitwiseCount` should override `prettyName` method.

### Why are the changes needed?
Improve the implement of some expression.

### Does this PR introduce any user-facing change?
 'Yes'. This PR will let user see the correct alias in schema.

### How was this patch tested?
Jenkins test.

### CODE REVIEW

**Comment by beliefer** [comment]
cc @HyukjinKwon

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala [hunk 1]
```diff
@@ -43,6 +43,8 @@ case class UnaryMinus(child: Expression) extends UnaryExpression
 
   override def toString: String = s"-$child"
 
+  override def prettyName: String = "-"
```
Comment (beliefer) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala:46]: @HyukjinKwon Should we change "-" to "negative" ?

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala:46]: I think it should be `negative` but should show `-` when it's used as the sql operator. You should be able to do this with if-else. e.g.)

```
getTagValue(FunctionRegistry.FUNC_ALIAS).getOrElse("-")
```

You might need to override `sql` IIRC.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/MaxByAndMinBy.scala [hunk 2]
```diff
@@ -100,6 +100,9 @@ abstract class MaxMinBy extends DeclarativeAggregate {
   """,
   since = "3.0.0")
 case class MaxBy(valueExpr: Expression, orderingExpr: Expression) extends MaxMinBy {
+
+  override def prettyName: String = "max_by"
+
   override protected def funcName: String = "max_by"
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/MaxByAndMinBy.scala:106]: Seems like you can remove this and replace it to `prettyName`.

Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/MaxByAndMinBy.scala:106]: Yes. I will replace it.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala (hunk 1 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala]: OK.

**Comment by beliefer** [comment]
retest this please

**Comment by beliefer** [comment]
cc @HyukjinKwon

**Review by HyukjinKwon — APPROVED** [review]

**Comment by beliefer** [comment]
cc @gatorsmile

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Comment by beliefer** [comment]
@HyukjinKwon Thanks!

**Pull request closed without merging**